pynxtools_xrd/reader.py

```diff
@@ -19,162 +19,115 @@
+from fairmat_readers_xrd import read_file, ureg
 from pynxtools.dataconverter.readers.base.reader import BaseReader
-
-CONVERT_DICT: Dict[str, str] = {
-    "unit": "@units",
-    "Instrument": "INSTRUMENT[instrument]",
-    "Source": "SOURCE[source]",
-    "Detector": "DETECTOR[detector]",
-    "Collection": "COLLECTION[collection]",
-    "Sample": "SAMPLE[sample]",
-    "version": "@version",
-    "User": "USER[user]",
-}
-
-
-# Global var to collect the root from get_template_from_nxdl_name()
-# and use it in the the the varidate_data_dict()
-ROOT: ET.Element = None
-REPLACE_NESTED: Dict[str, Any] = {}
-XRD_FILE_EXTENSIONS = [".xrdml", "xrdml", ".udf", ".raw", ".xye"]
-
-
-def get_template_from_nxdl_name(nxdl_name):
-    """Generate template from nxdl name.
-
-    Example of nxdl name could be NXxrd_pan.
-    Parameters
-    ----------
-    nxdl_name : str
-        Name of nxdl file e.g. NXmpes
-
-    Returns
-    -------
-    Template
-        Empty template.
-
-    Raises
-    ------
-    ValueError
-        Error if nxdl file is not found.
-    """
-    nxdl_file = nxdl_name + ".nxdl.xml"
-    current_path = Path(__file__)
-    def_path = current_path.parent.parent.parent.parent / "definitions"
-    # Check contributed defintions
-    full_nxdl_path = Path(def_path, "contributed_definitions", nxdl_file)
-    root = None
-    if full_nxdl_path.exists():
-        root = ET.parse(full_nxdl_path).getroot()
-    else:
-        # Check application definition
-        full_nxdl_path = Path(def_path, "applications", nxdl_file)
-
-    if root is None and full_nxdl_path.exists():
-        root = ET.parse(full_nxdl_path).getroot()
-    else:
-        full_nxdl_path = Path(def_path, "base_classes", nxdl_file)
-
-    if root is None and full_nxdl_path.exists():
-        root = ET.parse(full_nxdl_path).getroot()
-    elif root is None:
-        raise ValueError("Need correct NXDL name")
-
-    template = Template()
-    generate_template_from_nxdl(root=root, template=template)
-    return template
-
-
-def get_template_from_xrd_reader(nxdl_name, file_paths):
-    """Get filled template from reader.
-
-    Parameters
-    ----------
-    nxdl_name : str
-        Name of nxdl definition
-    file_paths : Tuple[str]
-        Tuple of path of files.
-
-    Returns
-    -------
-    Template
-        Template which is a map from NeXus concept path to value.
-    """
-
-    template = get_template_from_nxdl_name(nxdl_name)
-
-    data = XRDReader().read(template=template, file_paths=file_paths)
-    validate_data_dict(template=template, data=data, nxdl_root=ROOT)
-    return data
+from pynxtools.dataconverter.readers.json_map.reader import (
+    fill_undocumented,
+    fill_documented,
+)
 
 
 # pylint: disable=too-few-public-methods
 class XRDReader(BaseReader):
     """Reader for XRD."""
 
     supported_nxdls = ["NXxrd_pan"]
+    supported_formats = [".rasx", ".xrdml", ".brml"]
+
+    mapping = {
```

Comment:
That's also fine for me. I guess the advantage could be that we can customize this config.json in a Nomad Upload. But then it will require a bit more logic to differentiate between a config passed through the CLI/args vs the default fallback that we manually configure. 

Or is it okay if I just move the JSON out into a file and call it a day?

Reply:
Yeah, let's just put it to a file for now, I just don't like having this giant map in the middle of the code 👍

Reply:
This is done now.